### PR TITLE
Add create-dmg/1.0.10

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **create-dmg/1.0.10**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/create-dmg/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("create-dmg", dst="bin", src=self._source_subfolder)
+
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
```
Comment (uilianries) [inline recipes/create-dmg/all/conanfile.py:34]:
```suggestion

    def package_id(self):
        del self.info.settings.build_type
        del self.info.settings.compiler
        del self.info.settings.arch

```
remove settings here.

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by MartinDelille** [comment]
@uilianries I re-request your review because your first review was before the new policy.

**Comment by MartinDelille** [comment]
@uilianries I re-request your review because your first review was before the new policy.

**Comment by MartinDelille** [comment]
@jgsogo @uilianries Any remarks against this PR?

File: recipes/create-dmg/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def package_id(self):
+        del self.info.settings.build_type
+        del self.info.settings.compiler
+        del self.info.settings.arch
```
Comment (uilianries) [inline recipes/create-dmg/all/conanfile.py:39]:
```suggestion
        self.info.header_only() 
```
It's the same package content for any OS.

File: recipes/create-dmg/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from conans import ConanFile, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os"
```
Comment (uilianries) [inline recipes/create-dmg/all/test_package/conanfile.py:5]:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
```
Use full settings always

**Comment by MartinDelille** [comment]
@uilianries :

```
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Package for Header Only does not contain artifacts with these extensions: ['h', 'h++', 'hh', 'hxx', 'hpp'] (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014) 
```

**Comment by uilianries** [comment]
@MartinDelille I see. It will need an exceptional case. Let me open a PR to hooks. Thanks for pinging!

**Comment by uilianries** [comment]
It requires https://github.com/conan-io/hooks/pull/412 first.

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**